src/Symfony/Component/Workflow/DataCollector/WorkflowDataCollector.php

```diff
@@ -57,4 +66,115 @@ public function getWorkflows(): array
     {
         return $this->data['workflows'] ?? [];
     }
+
+    public function hash(string $string): string
+    {
+        return hash('sha256', $string);
+    }
+
+    private function getEventListeners(WorkflowInterface $workflow): array
+    {
+        $listeners = [];
+        $placeId = 0;
+        foreach ($workflow->getDefinition()->getPlaces() as $place) {
+            $eventNames = [];
+            $subEventNames = [
+                'leave',
+                'enter',
+                'entered',
+            ];
+            foreach ($subEventNames as $subEventName) {
+                $eventNames[] = sprintf('workflow.%s', $subEventName);
+                $eventNames[] = sprintf('workflow.%s.%s', $workflow->getName(), $subEventName);
+                $eventNames[] = sprintf('workflow.%s.%s.%s', $workflow->getName(), $subEventName, $place);
+            }
+            foreach ($eventNames as $eventName) {
+                // @phpstan-ignore-next-line
+                foreach ($this->eventDispatcher->getListeners($eventName) as $listener) {
+                    $listeners["place{$placeId}"][$eventName][] = $this->summarizeListener($listener);
+                }
+            }
+
+            ++$placeId;
+        }
+
+        foreach ($workflow->getDefinition()->getTransitions() as $transitionId => $transition) {
+            $eventNames = [];
+            $subEventNames = [
+                'guard',
+                'transition',
+                'completed',
+                'announce',
+            ];
+            foreach ($subEventNames as $subEventName) {
+                $eventNames[] = sprintf('workflow.%s', $subEventName);
+                $eventNames[] = sprintf('workflow.%s.%s', $workflow->getName(), $subEventName);
+                $eventNames[] = sprintf('workflow.%s.%s.%s', $workflow->getName(), $subEventName, $transition->getName());
+            }
+            foreach ($eventNames as $eventName) {
+                // @phpstan-ignore-next-line
+                foreach ($this->eventDispatcher->getListeners($eventName) as $listener) {
+                    $listeners["transition{$transitionId}"][$eventName][] = $this->summarizeListener($listener, $eventName, $transition);
+                }
+            }
+        }
+
+        return $listeners;
+    }
+
+    private function summarizeListener(callable $callable, string $eventName = null, Transition $transition = null): array
+    {
+        $extra = [];
+
+        if ($callable instanceof \Closure || \is_string($callable)) {
+            $r = new \ReflectionFunction($callable);
+            $title = (string) $r;
+        } else  if (\is_object($callable) && method_exists($callable, '__invoke')) {
+            $r = new \ReflectionMethod($callable, '__invoke');
+            $title = \get_class($callable) . '::__invoke()';
+        } elseif (\is_array($callable)) {
+            if ($callable[0] instanceof GuardListener) {
```

Comment:
Will this work fine if the dispatcher is traceable ?

Reply:
I don't know how, but yes.
I already get the traceable dispatcher 

![image](https://github.com/symfony/symfony/assets/408368/077b669d-ea19-4770-8036-c97950f653a9)